mlog/stdlog.go

```diff
@@ -72,3 +73,15 @@ func getStdLogOption() zap.Option {
 		},
 	)
 }
+
+type loggerWriter struct {
+	logFunc func(msg string, fields ...Field)
+}
+
+func (l *loggerWriter) Write(p []byte) (int, error) {
```

Comment:
I like it! Worth doing this for fully structured logs as well?

Can we connect the logs together with an identifier, even randomly generated? Or would we at least consider logging the events all with the same timestamp? This might be useful if we wanted to recover stack dumps, etc.

Reply:
Ya, I thought about that as a v2. I think that would be an interesting way to give more complicated plugins the freedom to use whatever structured logging package they please. 

go-plugin actually implements this, but only specifically for their own hclog library: https://github.com/hashicorp/go-plugin/blob/master/client.go#L766 If we do choose to implement that, might be a good idea to just generalize the upstream implementation.

Not sure what you mean by connecting the logs... these only appear in the Mattermost server logs and nowhere else. They appear with the plugin's ID so they can be identified. (see the above examples)

Reply:
I meant that if two, multi-line messages were logged concurrently, there's a chance that their respective lines might get interleaved and it could be hard to untangle them. I suspect it's not a blocker at the moment, but I can see this coming up.